src/components/login/login.js

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from 'react';
+import './login.scss';
+
+function Login({ onSuccess, error }) {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const handleLogin = (event) => {
+    event.preventDefault();
+    onSuccess(email, password);
+  };
+
+  return (
+    <form className="login" method="post" onSubmit={handleLogin}>
+      <legend>Login</legend>
+      {error ? <p>{error}</p> : null}
+      <fieldset>
+        <div className="field">
+          <label htmlFor="email">Email:</label>
+          <input
+            type="text"
+            name="email"
+            id="email"
+            value={email}
+            onChange={(e) => {
```

Comment:
There could be a performance issue since the function is defined during render. I believe this is less of an issue for functional due to not having to bind this.

Might be a premature optimisation to change at this point though.

(This code has already been changed anyway, so I'll resolve this one)